sklearn/preprocessing/tests/test_data.py

```diff
@@ -2224,3 +2225,51 @@ def test_quantile_transform_valid_axis():
 
     assert_raises_regex(ValueError, "axis should be either equal to 0 or 1"
                         ". Got axis=2", quantile_transform, X.T, axis=2)
+
+
+def test_boxcox_transformer():
+    bct = BoxCoxTransformer()
+    X = np.abs(X_1col)
+    assert_raises(NotFittedError, bct.transform, X)
+    assert_raises(NotFittedError, bct.inverse_transform, X)
+
+    # 1D test
+    X_trans = bct.fit_transform(X)
+    X_expected, lambda_expected = stats.boxcox(X.flatten())
+
+    assert_almost_equal(X_expected.reshape(-1, 1), X_trans)
+    assert_almost_equal(X, bct.inverse_transform(X_trans))
+    assert_almost_equal(lambda_expected, bct.lambdas_[0])
+    assert_equal(len(bct.lambdas_), X.shape[1])
+    assert_true(isinstance(bct.lambdas_, np.ndarray))
+
+    # 2D test
+    X = np.abs(X_2d)
+    X_trans = bct.fit_transform(X)
+    for j in range(X_trans.shape[1]):
+        X_expected, lmbda = stats.boxcox(X[:, j].flatten())
+        assert_almost_equal(X_trans[:, j], X_expected)
+        assert_almost_equal(lmbda, bct.lambdas_[j])
+    assert_equal(len(bct.lambdas_), X.shape[1])
+    assert_true(isinstance(bct.lambdas_, np.ndarray))
+
+    # Exceptions should be raised for negative arrays and zero arrays
+    X_with_negatives = X_2d
+    assert_raises(ValueError, bct.transform, X_with_negatives)
```

Comment:
`assert_raise_message` or `assert_raises_regexp` is preferred to ensure we're hitting the right error.